pennylane/measurements/measurements.py

```diff
@@ -162,6 +163,9 @@ def __init__(
             # Cast sequence of measurement values to list
             self.mv = obs if getattr(obs, "name", None) == "MeasurementValue" else list(obs)
             self.obs = None
+        elif is_abstract(obs):  # Catalyst program with qml.sample(m, wires=i)
```

Comment:
Giving a heads up that with the program capture project, `obs` will become abstract, so this will no longer be a valid way to separate between accepting an `observable` and a mid circuit measurement.

Reply:
Hmm good point, we can be more specific like asking if it's a jax tracer of type bool? We could also ask whether `qjit` is active but presumably that will be the case as well in the future with the program capture.

Thinking about though, with the program capture project I don't expect these patches to be required anymore anyways. The tracing part will be completely supplanted without Catalyst-specific differences (like the measurement value),  so we probably don't need to work around this.

Reply:
```suggestion
        elif is_abstract(obs):  # Catalyst program with qml.sample(m)
```

Reply:
When necessary, I will be able to differentiate between the two by check `obs.aval`.  No longer a concern.